src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -442,34 +510,286 @@ public int getID()
+        {
+            this.triggerAchievement(ModAchievements.achievementBuildOneHut);
+        }
+        if (Huts >= NUM_ACHIEVEMENT_SECOND)
+        {
+            this.triggerAchievement(ModAchievements.achievementBuild25Huts);
+        }
+        if (Huts >= NUM_ACHIEVEMENT_THIRD)
+        {
+            this.triggerAchievement(ModAchievements.achievementBuild100Huts);
+        }
+        if (Huts >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementBuild500Huts);
+        }
+        if (Huts >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementBuild1000Huts);
+        }
+    }
+    /**
+     * Increment the fish caught by this colony.
+     * Will award achievements for fish caught.
+     */
+    public void incrementFishCaught()
+    {
+        caughtFish++;
+        final int Fish = this.getCaughtFish();
+        if (Fish >= NUM_ACHIEVEMENT_FIRST)
+        {
+            this.triggerAchievement(ModAchievements.achievementCatchOneFish);
+        }
+        if (Fish >= NUM_ACHIEVEMENT_SECOND)
+        {
+            this.triggerAchievement(ModAchievements.achievementCatch25Fish);
+        }
+        if (Fish >= NUM_ACHIEVEMENT_THIRD)
+        {
+            this.triggerAchievement(ModAchievements.achievementCatch100Fish);
+        }
+        if (Fish >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementCatch500Fish);
+        }
+        if (Fish >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementCatch1000Fish);
+        }
+    }
+    /**
+     * Increment the carrots harvested by this colony.
+     * Will award achievements for carrots harvested.
+     */
+    public void incrementCarrotsHarvested()
+    {
+        harvestedCarrots++;
+        final int Carrots = this.getHarvestedCarrots();
+        if (Carrots >= NUM_ACHIEVEMENT_FIRST)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvestOneCarrot);
+        }
+        if (Carrots >= NUM_ACHIEVEMENT_SECOND)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest25Carrots);
+        }
+        if (Carrots >= NUM_ACHIEVEMENT_THIRD)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest100Carrots);
+        }
+        if (Carrots >= NUM_ACHIEVEMENT_FOURTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest500Carrots);
+        }
+        if (Carrots >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            this.triggerAchievement(ModAchievements.achievementHarvest1000Carrots);
+        }
+    }
+    /**
+     * Increment the potatoes harvested by this colony.
+     * Will award achievements for potatoes harvested.
+     */
+    public void incrementPotatoesHarvested()
+    {
+        harvestedPotatoes++;
+        final int Potatoes = this.getHarvestedPotatoes();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00117)